src/dev/display/st7789/lv_st7789.c

```diff
@@ -0,0 +1,363 @@
+/**
+ * @file lv_st7789.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_st7789.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+typedef struct {
+	lv_display_t			*disp;			/* the associated LVGL display object */
+	lv_st7789_send_cmd_cb_t	send_cmd;		/* platform-specific implementation to send a command to the LCD controller */
+	lv_st7789_send_cmd_cb_t	send_color;		/* platform-specific implementation to send pixel data to the LCD controller */
+    uint16_t				x_gap;			/* x offset of the (0,0) pixel in VRAM */
+    uint16_t				y_gap;			/* y offset of the (0,0) pixel in VRAM */
+    uint8_t 				madctl_reg;		/* current value of MADCTL register */
+    uint8_t 				colmod_reg;		/* current value of COLMOD register */
+} lv_st7789_driver_t;
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static void send_cmd(lv_st7789_driver_t *drv, uint8_t cmd, uint8_t *param, size_t param_size);
+static void send_color(lv_st7789_driver_t *drv, uint8_t cmd, uint8_t *param, size_t param_size);
+static void init(lv_st7789_driver_t * drv);
+static void set_onoff(lv_st7789_driver_t * drv, bool on);
+static void set_mirror(lv_st7789_driver_t * drv, bool mirror_x, bool mirror_y);
+static void set_swap_xy(lv_st7789_driver_t * drv, bool swap);
+static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map);
+static void res_chg_event_cb(lv_event_t * e);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+inline lv_st7789_driver_t *lv_st7789_get_driver(lv_display_t *disp)
```

Comment:
```suggestion
static inline lv_st7789_driver_t *lv_st7789_get_driver(lv_display_t *disp)
```

And please move it to the static functions